src/socketio.quiver.js

```diff
@@ -966,6 +966,56 @@ QuiverSocialProvider.prototype.addFriend_ = function(servers, userId, pubKey,
   return this.syncConfiguration_().then(function() {
     // TODO: Connect to all servers.
     return this.connectAsClient_(friendDesc, servers[0]).ready;
+  }.bind(this)).catch(function(err) {
+    // The connection to the invitation server has failed.  This could be
+    // because the front domain is malfunctioning.  Try again with one of the
+    // default front domains.  This last-ditch effort is worthwhile because
+    // (1) invitations only contain a single server, so the front domain is a
+    // single point of failure, and (2) there is a known related bug, #2619.
+    var server = servers[0];
+    if (!server.front) {
+      // This isn't a fronted server, so there's nothing to retry.
+      throw err;
+    }
+    var domainComponents = server.domain.split('.');
+    if (domainComponents.length < 3) {
+      // For very short domains, we don't know how to identify which CDN they
+      // use.
+      throw err;
+    }
+    var cloudDomain = domainComponents.slice(1).join('.');
+    var alternateFront = null;
+    // Find the first front domain for this cloud in the default set
+    for (var i = 0; i < QuiverSocialProvider.DEFAULT_SERVERS_.length; ++i) {
+      var defaultServer = QuiverSocialProvider.DEFAULT_SERVERS_[i];
+      if (defaultServer.domain.endsWith(cloudDomain) &&
+          defaultServer.type === server.type &&
+          defaultServer.front) {
+        alternateFront = defaultServer.front;
+        break;
+      }
+    }
+    if (!alternateFront) {
+      // We didn't find any default server on the same cloud, so we don't know
+      // of any alternate front domain.
+      throw err;
+    }
+    if (alternateFront === server.front) {
+      // The front we found was the same one we started with.  This will happen
+      // whenever the connection failure is due to a non-front-related network
+      // problem.  The deterministic ordering of DEFAULT_SERVERS, combined with
+      // this check, avoids an unbounded retry loop.
```

Comment:
Can we put a little more detail here?  Maybe mention that because all DEFAULT_SERVERS should currently have working front domains, and we expect at a0.awsstatic.com will continue working, we think that the way this should work is:
1. the user received a bad front domain in an invite which failed
2. they retry with the a0.awsstatic.com as the front domain
3. if there is another error, the code will select a0.awsstatic.com as the alternateFront again, and this ```alternateFront === server.front``` check will be hit.  Given that we expect a0.awsstatic.com should always work, this indicates an error that is not related to the front domain.